api/v1alpha1/proxmoxcluster_types.go

```diff
@@ -71,6 +72,11 @@ type ProxmoxClusterSpec struct {
 	// in the configuration and cloning of a proxmox VM. Multiple types of nodes can be specified.
 	// +optional
 	CloneSpec *ProxmoxClusterCloneSpec `json:"cloneSpec,omitempty"`
+
+	// CredentialsRef is a reference to a Secret that contains the credentials to use for provisioning this cluster. If not
+	// supplied then the credentials of the controller will be used.
+	// +optional
+	CredentialsRef *corev1.ObjectReference `json:"credentialsRef,omitempty"`
```

Comment:
Maybe `SecretReference` or if you expect the secret to be in the same namespace `LocalObjectReference` would be sufficient?

Reply:
SecretReference is okay,